dlt/load/load.py

```diff
@@ -556,12 +566,29 @@ def run(self, pool: Optional[Executor]) -> TRunMetrics:
         schema = self.load_storage.normalized_packages.load_schema(load_id)
         logger.info(f"Loaded schema name {schema.name} and version {schema.stored_version}")
 
+        # prepare load package state context
```

Comment:
i am injecting the load package state context here, i think it makes the most sense but it also clutters the load class a bit, wdyt? For now we only use the load package state for the destinations, so there is no need to implement this any other stage than load.

Reply:
Now when I look at this I think we do not need to inject state at all, what we could inject is load id and instance of the package storage as a context available to a function decorated with sink. we can do that later.